raw_bsd.go

```diff
@@ -192,6 +200,73 @@ func (p *packetConn) SetPromiscuous(b bool) error {
 	return syscall.SetBpfPromisc(p.fd, m)
 }
 
+// ifReq is the BSD struct passed to ioctl for multicast add/drop operations.
+type ifReq struct {
+	ifrName [unix.IFNAMSIZ]uint8
+	ifrAddr unix.SockaddrDatalink
+}
+
+const etherAdderAlen = 6
+
+// JoinMulticast joins or drops membership of data link layer multicast address on interface,
+// allowing it to recieve multicast traffic.
+func (p *packetConn) SetMulticast(b bool, addr net.HardwareAddr) error {
+	// Convert hardware address back to byte array form
+	var baddr [8]byte
+	copy(baddr[:], addr)
+
+	s, err := syscall.Socket(unix.AF_INET, unix.SOCK_DGRAM, unix.IPPROTO_UDP)
```

Comment:
Yep, that approach makes sense to me. If BSD is going to be a non-trivial undertaking, I'd be fine with leaving the BSD implementation stubbed for now and just doing Linux in this PR.